packages/devtools_app/lib/src/charts/chart.dart

```diff
@@ -0,0 +1,588 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+import '../auto_dispose_mixin.dart';
+import 'chart_controller.dart';
+import 'chart_trace.dart';
+
+class Chart extends StatefulWidget {
+  Chart(this.controller, String title) {
+    controller.title = title;
+  }
+
+  final ChartController controller;
+
+  @override
+  ChartState createState() => ChartState();
+}
+
+class ChartState extends State<Chart> with AutoDisposeMixin {
+  ChartController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = widget.controller;
```

Comment:
don't cache the controller at all or update the cached version when the widget changes.

Reply:
No reason, must have something I had played with originally and didn't remove.